gamemodes/terrortown/gamemode/client/cl_player_ext.lua

```diff
@@ -294,6 +307,9 @@ net.Receive("TTT2NotifyPlayerReadyOnClients", function()
     -- @realm shared
     -- stylua: ignore
     hook.Run("TTT2PlayerReady", ply)
+
+    -- Cache avatars for all players currently on the server (with the new player included)
+    CacheAllPlayerAvatars()
```

Comment:
Could you add the `ply` as a parameter to that function? There is no point in caching every player every time imho.

The function would then look like this:
```lua
local function CacheAllPlayerAvatars(ply)
    local plys = IsPlayer(ply) and {ply} or player.GetAll()

    for i = 1, #plys do
        local plyid64 = plys[i]:SteamID64()
        draw.RefreshAvatars(plyid64)
    end
end
```

Reply:
Ok, but be aware that I first tested it with only
```
local plyid64 = ply:SteamID64()
draw.RefreshAvatars(plyid64)
```
here before and it didn't work. Then I used the `CacheAllPlayerAvatars()` function here too to get it to work.

I didn't test your feedback but if you are sure that it works then it's good to go.

Reply:
Hm, I'm not sure that it works, I thought it would work though. That's a bit weird then

Reply:
Nvm, tested it with a friend and it works.